src/routes.js

```diff
@@ -1,32 +1,77 @@
-import React from 'react';
+/* eslint-disable */
+import React, { Component } from 'react';
 import PropTypes from 'prop-types';
 import { TransitionGroup, CSSTransition } from 'react-transition-group';
 import { Route, Redirect, Switch } from 'react-router-dom';
 import { Protocol, Start } from './containers';
 import { getCSSVariableAsNumber } from './utils/CSSVariables';
+import tween from './behaviours/Tweened/tween';
 
-const routes = ({ location }) => (
-  <TransitionGroup component={null}>
-    <CSSTransition appear timeout={getCSSVariableAsNumber('--animation-duration-fast-ms')} classNames="route" key={location.key}>
-      <Switch location={location}>
-        <Route
-          exact
-          path="/edit/:protocol"
-          component={Protocol}
-        />
-        <Route
-          exact
-          path="/"
-          component={Start}
-        />
-        <Redirect to="/" />
-      </Switch>
-    </CSSTransition>
-  </TransitionGroup>
-);
+const dispatchRouteAnimations = ({ pathname }) => {
+  switch (true) {
+    case /^\/$/.test(pathname):
+      tween({ name: "foo", from: "overview-panel", to: "protocol-stack" });
+      break;
+    case /^\/edit\/[^\/]+$/.test(pathname):
+      tween({ name: "foo", from: "protocol-stack", to: "overview-panel" });
+      break;
+  }
+}
 
-routes.propTypes = {
+class Routes extends Component {
+  componentDidMount() {
+    this.props.history.listen(dispatchRouteAnimations);
+  }
+
+  render() {
+    const { location } = this.props;
+    return (
+      <TransitionGroup component={null}>
+        <CSSTransition appear timeout={getCSSVariableAsNumber('--animation-duration-fast-ms')} classNames="route" key={location.key}>
```

Comment:
So this means that all screen transitions/tweens need to have the same duration (and have a tween, period)? That's fine, if that's the design goal. Just curious...

Reply:
Because we are using cloned elements for the tween they *could* differ, though more than likely we won't want that. Tween takes an argument for durations, so I should probably update this so that they relate somehow...